### ISSUE TITLE
Write support double precision IDF files

### ISSUE THREAD

**Issue by Manangka** [Deltares/imod-python#221]
In GitLab by @jc-hunink on Jan 11, 2022, 16:22

Reading of Double precision IDF files is supported. Saving an float64 xarray to idf, a single precision idf file is saved.

`da64=imod.idf.open([DP_idf.IDF](https://gitlab.com/deltares/imod/imod-python/uploads/aa13a72fe6007dff5ef67ee3d50df670/DP_idf.IDF))`

`print(da64.dtype)`

`imod.idf.save('temp.idf',da64)`

**Comment by Manangka** [comment]
In GitLab by @Huite on Jan 11, 2022, 22:19

This is already possible by specifying the dtype argument in the `save` call:

```python
imod.idf.save("temp", da, dtype=np.float64)
```

Automatically writing to 64bit IDF when the DataArray dtype is 64bit is a recipe for pain because of haphazard support of 64bit IDFs in general -- e.g. I ran into an issue with iMOD-WQ a while ago (which may have been fixed). Note that Python defaults to 64 bit floats on our machines, so changing it to respect the Python type will result in 64 bit IDFs almost exclusively.

Saving to a 64bit IDF is (in my opinion) a risky choice because IDF-IO is not consistent between all iMOD products (iMODFLOW/iMOD GUI/iMOD-WQ). These products can be relied on to work with 32bit IDF, so 32bit should remain the default.
